gateway/src/model.rs

```diff
@@ -586,6 +602,68 @@ impl<'de> Deserialize<'de> for CredentialLocation {
     }
 }
 
+
+pub enum Credential {
+    Static(SecretString),
+    FileContents(SecretString),
+    Dynamic(String),
+    None,
+    #[cfg(any(test, feature = "e2e_tests"))]
+    Missing,
+}
+
+
+impl TryFrom<(CredentialLocation, &str, &str)> for Credential {
+    type Error = Error;
+    #[allow(unused_variables)]
+    fn try_from((location, provider_type, model_name): (CredentialLocation, &str, &str)) -> Result<Self, Self::Error> {
+        match location {
+            CredentialLocation::Env(key_name) => {
+                match env::var(key_name) {
+                    Ok(value) => Ok(Credential::Static(SecretString::from(value))),
+                    Err(_) => {
+                        #[cfg(any(test, feature = "e2e_tests"))]
+                        {
+                            warn!("You are missing the credentials required for `model.{}.provider.{}`, so the associated tests will likely fail.",model_name,provider_type);
```

Comment:
this is a bit confusing as we use dotted paths like these for config paths (e.g. something like `models.gpt35.openai.gpt-35-turbo-0...`). Really what we want here is something like [serde_path_to_error](https://docs.rs/serde_path_to_error/latest/serde_path_to_error/) but there's a [bug](https://github.com/tensorzero/tensorzero/issues/335) there. 

I'd suggest we do this as "You are missing the credentials required for a {provider_type} model, so the associated tests will likely fail", just so that we're not misleading people and this works in general. So it would be 

```rust
impl TryFrom<(CredentialLocation, &str)> for Credential {
    type Error = Error;
    #[allow(unused_variables)]
    fn try_from((location, provider_type): (CredentialLocation, &str)) -> Result<Self, Self::Error> {
...
```

Thanks!